### DISCUSSION TITLE
Multiple binaries share the same codesign path

### DISCUSSION THREAD

**Discussion by ronyfadel** [sparkle-project/Sparkle#2650] [Q&A]
### Description of the problem
When trying to archive my app, I get this error, which happens in both SPM and manually copying Sparkle.framework

<img width="755" alt="Screenshot 2024-10-24 at 11 06 13 PM" src="https://github.com/user-attachments/assets/ae2edb4a-ea05-4c0d-b87e-00a45c8b45f6">

### Do you use Sandboxing in your app?
The parent app is sandboxed, but Sparkle is invoked by a LoginItem contained in the parent app bundle that is not sandboxed.

### Version of `Sparkle.framework` in the latest version of your app
2.6.4

### Steps to reproduce the behavior
1- be on Xcode 16.0
2- add Sparkle (Embed & Sign) in LoginItem
3- Archive
4- attempt to notarize

**Comment by zorgiepoo** [comment]
Were the symlinks in the framework properly preserved as the error message seems to otherwise indicate? There should be only one `Sparkle` binary in the framework, the others should be symlinks.

Also I am not sure Sparkle should be embedded in a LoginItem inside your app bundle. That sounds something Service Management should manage updating if your app is updated. Although, I haven't read up on this subject. (How do you go about updating your main app?)

**Comment by ronyfadel** [comment]
Hi @zorgiepoo ! 
> the others should be symlinks
I would've expected that.. that they'd be symlinks to the "B" version, but I'm seeing 3 copies of included binaries, one in the main framework folder, one in B and one in Current. Odd indeed!

> Also I am not sure Sparkle should be embedded in a LoginItem inside your app bundle
The login item updates the whole app bundle. That is because the app itself is barely used/run. The login item is a background daemon that does all the magic.
It had worked flawlessly for over 5 years now. The main issue AFAIK is the symlinks not being generated, but I don't fully understand why.

**Reply by zorgiepoo** [reply]
Okay. Missing symlinks is *usually* because of using a custom post run script that copies the framework incorrectly (certain commands that should be avoided like `cp -r` will/can do this). It is not usually because of Xcode doing it on your behalf, and that is what Xcode is trying to tell you. You may be able to look at / export the build log in full detail and see what steps Xcode is doing to produce your app bundle.

**Reply by ronyfadel** [reply]
Hi @zorgiepoo ! I didn't have any custom run scripts of my own, but I was able to fix the issue by re-copying Sparkle properly with a script.

fyi, Xcode was doing this:

```
Copy /Users/ronyfadel/Library/Developer/Xcode/DerivedData/Batteries-biwrcxizazeowebocntwqcyydwvx/Build/Products/Debug/io.fadel.Batteries.Helper.app/Contents/Frameworks/Sparkle.framework /Users/ronyfadel/Library/Developer/Xcode/DerivedData/Batteries-biwrcxizazeowebocntwqcyydwvx/SourcePackages/artifacts/sparkle/Sparkle/Sparkle.xcframework/macos-arm64_x86_64/Sparkle.framework (in target 'BatteriesHelper' from project 'Batteries')
    cd /Users/ronyfadel/Developer/Batteries/Batteries
    builtin-copy -exclude .DS_Store -exclude CVS -exclude .svn -exclude .git -exclude .hg -exclude Headers -exclude PrivateHeaders -exclude Modules -exclude \*.tbd -resolve-src-symlinks -remove-static-executable /Users/ronyfadel/Library/Developer/Xcode/DerivedData/Batteries-biwrcxizazeowebocntwqcyydwvx/SourcePackages/artifacts/sparkle/Sparkle/Sparkle.xcframework/macos-arm64_x86_64/Sparkle.framework /Users/ronyfadel/Library/Developer/Xcode/DerivedData/Batteries-biwrcxizazeowebocntwqcyydwvx/Build/Products/Debug/io.fadel.Batteries.Helper.app/Contents/Frameworks
```

**Reply by zorgiepoo** [reply]
If you feel like it's an Xcode bug, I would report a feedback assistant report to Apple including a sample/repro case and build log. In the past, they have got around fixing bugs like this.
